### DISCUSSION TITLE
ESP3D V3 for ESP8266 error on building

### DISCUSSION THREAD

**Discussion by Stephano2000** [luc-github/ESP3D#854] [Questions]
Hi,
As it's  the 1st time I build ESP3D V3 for ESP8266 (I have built V3 for ESP32 and V2.1 for ESP8266 successfully), I am not sure again if I am missing something in the build process. I get the following output/error:

`In file included from sketch/src/core/benchmark.cpp:21:0:
sketch/src/core/../include/esp3d_config.h:36:2: error: #error "No configuration file found"
 #error "No configuration file found"
  ^
In file included from sketch/src/core/../include/esp3d_config.h:42:0,
                 from sketch/src/core/benchmark.cpp:21:
sketch/src/core/../include/../include/sanity_esp3d.h:34:2: error: #error Choose Preferences or EEPROM for settings
 #error Choose Preferences or EEPROM for settings
  ^
sketch/src/core/../include/../include/sanity_esp3d.h:55:2: error: #error ESP_SERIAL_OUTPUT must be defined
 #error ESP_SERIAL_OUTPUT must be defined
  ^
sketch/src/core/../include/../include/sanity_esp3d.h:64:2: error: #error Serial 2 is not available in ESP8266
 #error Serial 2 is not available in ESP8266
  ^
exit status 1
Error compiling for board Generic ESP8266 Module.
`

Bug or PEBCAK?

**Comment by luc-github** [comment]
what is your configurtion.h? looks there is none

**Reply by Stephano2000** [reply]
Indeed. I used the configurator to get the file and I pasted it over the ESP32 configuration.h (which builds fine).
That's what left me stumped.

Do you want me to post configuration.h here?

**Reply by luc-github** [reply]
yes I am still outside but can have a look

**Reply by Stephano2000** [reply]
I had to  change the extension from .h to .txt to be able to attach configuration.h:
[configuration.h](https://github.com/luc-github/ESP3D/files/10535767/configuration.txt)

**Reply by luc-github** [reply]
nothing pop to my eyes - I will test it later

**Reply by luc-github** [reply]
I have tested using your configuration.h and no issue on my side :
platformio 
![image](https://user-images.githubusercontent.com/8822552/215498330-5c28347f-2dd4-4186-852b-e6fe452f4f71.png)
Arduino ide + 3.0.2 core because git has a bug with esp8265 currently

![image](https://user-images.githubusercontent.com/8822552/215501277-9e4b0df7-3696-4e5e-952f-0a8d4554c84a.png)

**Reply by Stephano2000** [reply]
i will test on a windows machine with an updated Arduino IDE and report back.
I am currently on 1.8.19 (stock Ubuntu).
Btw, I can't locate a Reset button in the configurator.

**Reply by luc-github** [reply]
reset button in configurator is `F5` refresh the page reset all

**Reply by luc-github** [reply]
I am on windows so could be a case issue  but the github -ci run on linux and it is pass, so I do not think it is the problem
![image](https://user-images.githubusercontent.com/8822552/215506401-d2eb8fc1-4ea4-473a-b6ac-08644a0812f6.png)

**Comment by Stephano2000** [comment]
I am not sure what you mean by core 3.0.2, but I tried to build for ESP8266 on the ubuntu box:
ESP3D-3.0-alpha1: Success

ESP3D-3.0-alpha2: Failure
`In file included from sketch/src/core/benchmark.cpp:21:0:
sketch/src/core/../include/esp3d_config.h:31:32: fatal error: ../esp3dlib_config.h: No such file or directory
 #include "../esp3dlib_config.h"
                                ^
compilation terminated.
exit status 1
Error compiling for board Generic ESP8266 Module.`

ESP3D-3.0.0-alpha3: Failure
`In file included from sketch/src/core/benchmark.cpp:21:0:
sketch/src/core/../include/esp3d_config.h:36:2: error: #error "No configuration file found"
 #error "No configuration file found"
  ^
In file included from sketch/src/core/../include/esp3d_config.h:42:0,
                 from sketch/src/core/benchmark.cpp:21:
sketch/src/core/../include/../include/sanity_esp3d.h:34:2: error: #error Choose Preferences or EEPROM for settings
 #error Choose Preferences or EEPROM for settings
  ^
sketch/src/core/../include/../include/sanity_esp3d.h:55:2: error: #error ESP_SERIAL_OUTPUT must be defined
 #error ESP_SERIAL_OUTPUT must be defined
  ^
sketch/src/core/../include/../include/sanity_esp3d.h:64:2: error: #error Serial 2 is not available in ESP8266
 #error Serial 2 is not available in ESP8266
  ^
exit status 1
Error compiling for board Generic ESP8266 Module.`

ESP3D-3.0-master: Failure
`In file included from sketch/src/core/benchmark.cpp:21:0: sketch/src/core/../include/esp3d_config.h:36:2: error: #error "No configuration file found" #error "No configuration file found" ^ In file included from sketch/src/core/../include/esp3d_config.h:42:0, from sketch/src/core/benchmark.cpp:21: sketch/src/core/../include/../include/sanity_esp3d.h:34:2: error: #error Choose Preferences or EEPROM for settings #error Choose Preferences or EEPROM for settings ^ sketch/src/core/../include/../include/sanity_esp3d.h:55:2: error: #error ESP_SERIAL_OUTPUT must be defined #error ESP_SERIAL_OUTPUT must be defined ^ sketch/src/core/../include/../include/sanity_esp3d.h:64:2: error: #error Serial 2 is not available in ESP8266 #error Serial 2 is not available in ESP8266 ^ exit status 1 Error compiling for board Generic ESP8266 Module.`

I think there is something wrong in this section of esp3d_config.h (don't mind the larger font):
`#if defined __has_include
#  if __has_include ("../../configuration.h")
#include "../../configuration.h"
#define ESP3D_CODE_BASE "ESP3D"
#else
#undef DISABLED
#undef _BV
#  if __has_include ("../esp3dlib_config.h")
#include "../esp3dlib_config.h"
#define ESP3D_CODE_BASE "ESP3DLib"
#else
#error "No configuration file found"
#endif
#endif
#endif`

If this part is deleted except #include "../../configuration.h", compilation proceeds though it hits some other seemingly unrelated error:
`/tmp/arduino_build_565391/sketch/src/core/espcmd/ESP401.cpp: In member function 'bool Commands::ESP401(const char*, level_authenticate_type, ESP3DOutput*)':
/tmp/arduino_build_565391/sketch/src/core/espcmd/ESP401.cpp:89:34: error: ambiguous overload for 'operator=' (operand types are 'String' and 'bool')
                         response = false;
                                  ^
/tmp/arduino_build_565391/sketch/src/core/espcmd/ESP401.cpp:89:34: note: candidates are:
In file included from /home/supra/.arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/Arduino.h:243:0,
                 from /tmp/arduino_build_565391/sketch/src/core/espcmd/../../include/esp3d_config.h:23,
                 from /tmp/arduino_build_565391/sketch/src/core/espcmd/ESP401.cpp:20:
/home/supra/.arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/WString.h:95:18: note: String& String::operator=(const String&)
         String & operator =(const String &rhs);
                  ^
/home/supra/.arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/WString.h:96:18: note: String& String::operator=(const char*)
         String & operator =(const char *cstr);
                  ^
/home/supra/.arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/WString.h:97:18: note: String& String::operator=(const __FlashStringHelper*)
         String & operator = (const __FlashStringHelper *str);
                  ^
/home/supra/.arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/WString.h:99:18: note: String& String::operator=(String&&)
         String & operator =(String &&rval);
                  ^
/home/supra/.arduino15/packages/esp8266/hardware/esp8266/2.7.4/cores/esp8266/WString.h:100:18: note: String& String::operator=(StringSumHelper&&)
         String & operator =(StringSumHelper &&rval);
                  ^
Using library ESP8266WiFi at version 1.0 in folder: /home/supra/.arduino15/packages/esp8266/hardware/esp8266/2.7.4/libraries/ESP8266WiFi 
Using library ESP8266WebServer at version 1.0 in folder: /home/supra/.arduino15/packages/esp8266/hardware/esp8266/2.7.4/libraries/ESP8266WebServer 
Using library EEPROM at version 1.0 in folder: /home/supra/.arduino15/packages/esp8266/hardware/esp8266/2.7.4/libraries/EEPROM 
Using library LittleFS at version 0.1.0 in folder: /home/supra/.arduino15/packages/esp8266/hardware/esp8266/2.7.4/libraries/LittleFS 
Using library ESP8266SSDP at version 1.0 in folder: /home/supra/.arduino15/packages/esp8266/hardware/esp8266/2.7.4/libraries/ESP8266SSDP 
Using library ESP8266mDNS at version 1.2 in folder: /home/supra/.arduino15/packages/esp8266/hardware/esp8266/2.7.4/libraries/ESP8266mDNS 
Using library DNSServer at version 1.1.1 in folder: /home/supra/.arduino15/packages/esp8266/hardware/esp8266/2.7.4/libraries/DNSServer 
Using library arduinoWebSockets-2.3.6 at version 2.3.6 in folder: /home/supra/Arduino/libraries/arduinoWebSockets-2.3.6 
Using library Hash at version 1.0 in folder: /home/supra/.arduino15/packages/esp8266/hardware/esp8266/2.7.4/libraries/Hash 
exit status 1
Error compiling for board Generic ESP8266 Module.`

**Comment by Stephano2000** [comment]
Tested on Windows 10 with Arduino IDE 2.0.3. Same problem.

**Reply by luc-github** [reply]
as you use arduino you can chose which esp8266 core is used , you are using core 2.7.4 , for esp3d v3  I used 3.0.2

**Reply by luc-github** [reply]
and better use esp3d v3 git version , alpha 2 is outdated now

**Reply by Stephano2000** [reply]
I updated the ESP8266 board core from 2.7.4 to 3.0.2, and i was finally able to build ESP V3 (git version) for ESP8266.
Thank you!

**Reply by luc-github** [reply]
I suggest you to use platformio,I ensure compilation is working out of box without issue
